portal/components/MarkDownLoader.js

```diff
@@ -11,24 +12,68 @@ class MarkDownLoader extends React.Component {
     this.state = { markdown: '' }
   }
 
-  componentWillMount() {
-    fetch(this.props.file)
-      .then(response => {
-        return response.text()
-      })
-      .then(text => {
-        this.setState({ markdown: ReplaceKpnStyleBundleVersion(text) });
-      });
+  isMarkDownfile(source) {
+    const mdFileRegex = /\.md$/;
+    return mdFileRegex.test(source);
+  }
+
+  getMarkdownContent(source) {
+    if(this.isMarkDownfile(source)) {
+      fetch(source)
+        .then(response => {
+          return response.text()
+        })
+        .then(text => {
+          this.setState({ markdown: ReplaceKpnStyleBundleVersion(text) });
+        });
+    } else {
+      this.setState({ markdown: ReplaceKpnStyleBundleVersion(source) });
+    }
+  }
 
+  componentDidMount() {
+    this.getMarkdownContent(this.props.content);
   }
 
   render() {
     return (
-      <Content title={this.props.title}>
-        <ReactMarkdown className="kpn-style" renderers={{code: CodeBlock}} source={this.state.markdown} escapeHtml={false} />
+      <Content title={this.props.data?.category || this.props.title}>
+        {this.props.data && Array.isArray(this.props.data.content) &&
+          <>
+            <div className="jumbotron">
+              <div className="jumbotron__body">
+                <div className="jumbotron__title">Button</div>
+              </div>
+            </div>
+
+            <nav className="tab">
+              <ul className="tab__menu">
+                {
+                  this.props.data.content.map((content) =>
```

Comment:
You are not returning anything in this map function, right? So you could use a forEach instead

Reply:
Unfortunately forEach is not working for inline React templates